whatsapp_theming/lib/darkmode.dart

```diff
@@ -0,0 +1,177 @@
+import 'package:flutter/material.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'package:theme_provider/theme_provider.dart';
+
+class DarkModeTest extends StatefulWidget {
+  @override
+  _DarkModeTestState createState() => _DarkModeTestState();
+}
+
+class _DarkModeTestState extends State<DarkModeTest> {
+  int radiovalue = 0;
+  void radioValue(int val) {
+    setState(() {
+      radiovalue = val;
+    });
+  }
+
+  saveDarkMode(int value) async {
+    SharedPreferences prefs = await SharedPreferences.getInstance();
+    prefs.setBool('isUserNew', false);
+    prefs.setInt('radiovalue', value);
+  }
+
+  void _checkUserisNew() async {
+    SharedPreferences prefs = await SharedPreferences.getInstance();
+    try {
+      dynamic result = prefs.getBool('isUserNew');
+      if (result != null) {
+        radiovalue = prefs.getInt('radiovalue');
+      } else {
+        _checkDarkModeEnabled();
+      }
+    } catch (e) {
+      print(e.toString());
+    }
+  }
+
+  void _checkDarkModeEnabled() {
+    radiovalue = 2;
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    WidgetsBinding.instance.addPostFrameCallback((_) async {
+      _checkUserisNew();
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
Split your code to avoid nesting so much. Create parts into separate files and use it here. At this point this whole code is not readable, let alone reviewable.